src/nni_manager/core/nniDataStore.ts

```diff
@@ -160,25 +167,56 @@ class NNIDataStore implements DataStore {
 
     private async getFinalMetricData(trialJobId: string): Promise<any> {
         const metrics: MetricDataRecord[] = await this.getMetricData(trialJobId, 'FINAL');
-        if (metrics.length > 1) {
-            this.log.error(`Found multiple final results for trial job: ${trialJobId}`);
+
+        const multiPhase: boolean | undefined = await this.isMultiPhase();
+
+        if (metrics.length > 1 && !multiPhase) {
+            this.log.error(`Found multiple FINAL results for trial job ${trialJobId}`);
+        }
+
+        return metrics[metrics.length - 1];
+    }
+
+    private async isMultiPhase(): Promise<boolean|undefined> {
```

Comment:
Should change return type to Promise<boolean>, since we definitely return a boolean value according to line 185-189?

Reply:
Thanks, updated per comments.